src/compute/src/render/mod.rs

```diff
@@ -144,6 +147,31 @@ pub fn build_compute_dataflow<A: Allocate>(
     compute_state: &mut ComputeState,
     dataflow: DataflowDescription<Plan, CollectionMetadata>,
 ) {
+    fn intercept_source_instantiation_frontiers<G>(
+        source_instantiation: &Stream<G, (Row, mz_repr::Timestamp, Diff)>,
+        logger: &Logger,
+        source_id: &GlobalId,
+        index_ids: &Vec<GlobalId>,
+        sink_ids: &Vec<GlobalId>,
+    ) -> Stream<G, (Row, mz_repr::Timestamp, Diff)>
+    where
+        G: Scope<Timestamp = mz_repr::Timestamp>,
+    {
+        let logger = logger.clone();
+        let source_id = *source_id;
+        let mut dataflow_ids = index_ids.clone();
+        dataflow_ids.extend(sink_ids);
+        source_instantiation.inspect_container(move |event| {
+            if let Err(frontier) = event {
+                for time in frontier.iter() {
+                    for dataflow_id in dataflow_ids.iter() {
+                        logger.log(ComputeEvent::SourceFrontier(*dataflow_id, source_id, *time));
+                    }
+                }
```

Comment:
As you noted before, this is only correct as long as frontier only has a single element. Should we ever move away from totally ordered times, this code will continue to function but will produce nonsensical results.

I think it would be better if we crashed instead of silently producing wrong results. That way, we'll at least notice and know that this place needs to be adjusted. How about:
```rust
                if let [time] = frontier {
                    for dataflow_id in dataflow_ids.iter() {
                        logger.log(ComputeEvent::SourceFrontier(*dataflow_id, source_id, *time));
                    }
                } else {
                    panic!("only totally ordered times are supported");
                }
```

Reply:
I don't think the check is correct as we could encounter an empty frontier once the upstream operator ceases to hold capabilities. I'm not worried about supporting partially-ordered data because it'll have more implications than just this code -- even tracking corresponding frontier advancements suddenly becomes non-trivial, so we'd need to re-think the general approach.

Reply:
Works for me! I agree the loop might give a wrong impression on what kind of time we support here.